python/taichi/lang/matrix.py

```diff
@@ -522,13 +491,10 @@ def to_numpy(self, as_vector=False):
         ti.sync()
         return ret
 
-    def to_torch(self, as_vector=False, device=None):
+    def to_torch(self, device=None, keep_dims=False):
         import torch
-        if as_vector:
-            assert self.m == 1, "This matrix is not a vector"
-            dim_ext = (self.n, )
-        else:
-            dim_ext = (self.n, self.m)
+        as_vector = self.m == 1 and not keep_dims
+        dim_ext = (self.n, ) if as_vector else (self.n, self.m)
```

Comment:
Let's skip `self.m` as well if it value is `1`. `len(dim_ext)=`
 - `0` for `1x1` matrices
 - `1` for `nx1` or `1xn` matrices (`n != 1`)
 - `2` for `nxm` matrices (`n, m != 2`)

Reply:
Then I should change matrix_to_ext_arr to treat the useless `1x1` matrices?

Reply:
Yes, it seems that we need a `keep_dims` parameter in `matrix_to_ext_arr` as well.

Thanks!

Reply:
Special treating 1x1 mats sounds unreasonable to me..

Reply:
Few people would use 1x1 matrices anyway. Let's just make the behavior consistent: dimensionality of size `1` should be skipped.

Reply:
Wdym? Do we consider 1x1 mat as 1D vector?

Reply:
I mean, if the matrix is `1x1`, then the numpy array, when `keep_dims=False`, should have the same shape as the matrix shape, instead of matrix shape + `(1, 1)`.